### PR TITLE
Allow building without X11 dependencies

### PR DESCRIPTION
**Pull request by varigigi**
When compiling for a pure Wayland environment no X11 dependencies are
expected.
This patch ensures that all the X11 stuff is guarded appropriately.

Signed-off-by: Pierluigi Passaro <pierluigi.p@variscite.com>

### CODE REVIEW

File: build/cmake/toolkit.cmake [hunk 1]
```diff
@@ -70,7 +70,7 @@ set(wxTOOLKIT_LIBRARIES)
 set(wxTOOLKIT_VERSION)
 
 if(UNIX AND NOT APPLE AND NOT WIN32)
-    find_package(X11 REQUIRED)
+    find_package(X11)
```
Comment (vadz) [inline build/cmake/toolkit.cmake:73]: I wonder if we could keep it required when X11 support is enabled at GTK level. The trouble is that I don't really know how to check for this other than by running a test GTK program. Does anybody know of a better/simpler way by chance?

File: src/unix/utilsx11.cpp [hunk 2]
```diff
@@ -13,7 +13,9 @@
 // for compilers that support precompilation, includes "wx.h".
 #include "wx/wxprec.h"
 
+#if defined(__WXX11__) || (defined(__WXGTK__) && defined(GDK_WINDOWING_X11)) || defined(__WXMOTIF__)
```
Comment (vadz) [inline src/unix/utilsx11.cpp:16]: You will need to move this check below, `GDK_WINDOWING_X11` is definitely not defined here yet.

If you wait until #2261 is merged (or rebase your branch on top of it), you can just do it inside `#if wxHAS_X11_SUPPORT`.

**Review by vadz — COMMENTED** [review]
I think I see the problem now: X11 headers are just not available in your build, so we shouldn't include them at all, right?

File: build/cmake/toolkit.cmake (hunk 1 above)
Comment (swt2c) [reply build/cmake/toolkit.cmake:73]: We have a `wxHAVE_GDK_WAYLAND` for Wayland in FindGTK3.cmake, see below.  Something similar could be done for X11 support.

https://github.com/wxWidgets/wxWidgets/blob/master/build/cmake/modules/FindGTK3.cmake#L48

Comment (vadz) [reply build/cmake/toolkit.cmake:73]: Thanks, this is exactly what I was looking for! I'll add a similar `wxHAVE_GDK_X11`.

**Comment by vadz** [comment]
I've integrated the changes that should fix the problems you've encountered in #2261, please test it if you can. TIA!

**Pull request closed without merging**